src/armory/Armory.js

```diff
@@ -1,431 +1,517 @@
+						treads: getOptionsOfType(data.inventory.tankComponents, 'treads'),
+						items: getOptionsOfType(data.inventory.tankComponents, 'item'),
+					});
 				}
 			})
-		).catch(
-			error => {
-				console.log('Couldnt connect to server!');
-				console.log(error);
-			}
-		);
-		
-	};
+		)
+	}
 
+	// Converts camel case to title case.
+	toTitleCase(str: string): string {
+		// Check for undefined strings.
+		if(str == null) {
+			return '';
+		}
+
+		let newStr = str.replace( /([A-Z])/g, " $1");
+		return newStr.charAt(0).toUpperCase() + newStr.slice(1);
+	}
+
+	// Find the tank via its name and set it to the selectedTank and in a Cookie for Casus.
+	changeSelectedTank(newTankId: string): void {
+		// Null check for if the user has no tanks.
+		// We should probably make it impossible for users to have none.
+		if(this.state.allTanks == null) {
+			console.log('No tanks found!');
+			return;
+		}
 
-	//This handles the changes if a user changes tanks or its components
-	//Thsi has to be an any because this taget uses label which is not a part of HTMLElement. 
-	handleChangeInTankOptions = ({ target }:{target:HTMLInputElement}) => {
-		this.setState({ selectedTankId: target.value});
-		this.getSelectedTank();
+		// Find the new tank via its id and set it to selectedTank and a cookie.
+		const newTank: Tank = getTank(this.state.allTanks.find(tank => tank._id === newTankId));
+		this.setState({ selectedTank: newTank});
+		const cookies = new Cookies();
+		cookies.set('tank', newTank);
 	}
 
-	handleChangeInChassisOptions = ({ target }:{target:HTMLInputElement }) => {
-		this.setState({selectedChassis: target.value});
+	// Function that will save the selectedTank.
+	saveTank(): void {
+		// Null check for selectedTank.
+		if(this.state.selectedTank == null) {
+			console.log('No selected tank!');
+			return;
+		}
+
+		// Save tank.
+		fetch('/api/tank/tankUpdate/' + this.state.selectedTank._id, {
+			method: 'PUT',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true'
+			},
+			body: JSON.stringify({ 
+				tankName: this.state.selectedTank.tankName, 
+				userId: this.state.selectedTank.userId, 
+				components: this.state.selectedTank.components, 
+				casusCode: this.state.selectedTank.casusCode, 
+				isBot: false 
+			}),
+		});
 	}
 
-	handleChangeInWeaponOneOptions = ({ target }:{target:HTMLInputElement }) => {
-		this.setState({selectedWeaponOne: target.value});
+	// ALL UPDATES NEED API CALL TO UPDATE INVENTORY.
+	updateChassis(chassis: TankComponent): void {
+		// Set the newChassis.
+		const newChassis: Chassis = new Chassis(chassis);
+
+		// Update the selected tank with the new chassis.
+		if(this.state.selectedTank == null) {
+			console.log('No selected tank found!');
+			return;
+		}
+		// Setup a new tank that will be updated and set to the selected tank.
+		let updatedTank: Tank = getTank(this.state.selectedTank);
```

Comment:
todo: make const

Reply:
change to this copy the front end tank instead of using getTank, which turns a backend version into the front-end version